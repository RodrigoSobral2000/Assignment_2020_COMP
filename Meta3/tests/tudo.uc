void v(void b);
void g(void) {
    void l;
    return v();
}

int main(void x) {
    if (2) return 1;

    if (1) {
        return 5;
    } else {
        return 9;
    }
}