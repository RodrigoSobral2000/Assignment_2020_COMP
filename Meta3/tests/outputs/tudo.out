Line 1, col 8: Invalid use of void type in declaration
Line 3, col 10: Invalid use of void type in declaration
Line 4, col 12: Unknown symbol v
Line 4, col 12: Conflicting types (got undef, expected void)
Line 7, col 10: Invalid use of void type in declaration
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
g	void(void)

===== Function g Symbol Table =====
return	void

Program
..FuncDeclaration
....Void
....Id(v)
....ParamList
......ParamDeclaration
........Void
........Id(b)
..FuncDefinition
....Void
....Id(g)
....ParamList
......ParamDeclaration
........Void
....FuncBody
......Declaration
........Void
........Id(l)
......Return
........Call - undef
..........Id(v) - undef
..FuncDefinition
....Int
....Id(main)
....ParamList
......ParamDeclaration
........Void
........Id(x)
....FuncBody
......If
........IntLit(2)
........Return
..........IntLit(1)
........Null
......If
........IntLit(1)
........Return
..........IntLit(5)
........Return
..........IntLit(9)
